### DISCUSSION TITLE
Kubernetes 贡献指南 - 面向信仰编程 · /kubernetes-contributor

### DISCUSSION THREAD

**Discussion by draveness** [draveness/blog-comments#614] [Announcements]
https://draveness.me/kubernetes-contributor/

**Comment by 741369** [comment]
膜拜大佬的成长之路

**Comment by yuchanns** [comment]
因为太菜了只能找找文章语病
> 作者不仅从 Kubernetes 社区中不仅学习到了 Go 语言的开发规范...

**Comment by draveness** [comment]
> 因为太菜了只能找找文章语病
> 
> > 作者不仅从 Kubernetes 社区中不仅学习到了 Go 语言的开发规范...

多看看总能发现问题的

**Comment by jasonlift** [comment]
真心佩服

**Comment by chi1231** [comment]
每当自己学不下去了就来看看看博主的文章，真的可以让自己充满斗志，反省自己学的还不够水平，激励自己要成为博主这样的做技术的人

**Comment by sabercoding** [comment]
👍

**Comment by chenggui53** [comment]
请问大佬博客的文章能否增加调整行宽的选项，现在的行宽稍微有点小，导致电脑下每屏的显示效率有些低。

**Comment by draveness** [comment]
> 请问大佬博客的文章能否增加调整行宽的选项，现在的行宽稍微有点小，导致电脑下每屏的显示效率有些低。

我回头可以稍微调整一下，窄的行距，阅读的时候眼睛可以不用动...

**Comment by lixiang233** [comment]
我的第一个pr也是删了重复的代码  还是你review的

**Comment by draveness** [comment]
> 我的第一个pr也是删了重复的代码 还是你review的

哈哈哈哈，是这个吧 https://github.com/kubernetes/kubernetes/pull/90542

**Comment by lcwxz1989** [comment]
关注了很久，支持支持，真的不吝赐教。受益匪浅

**Comment by Cactusinhand** [comment]
向大佬致敬！
请问，像kubernetes这种特别流行特别活跃的社区，你是如何在一年之类取得社区成员的认可，成为这个社区的一员的呢？

**Comment by draveness** [comment]
> 向大佬致敬！
> 请问，像kubernetes这种特别流行特别活跃的社区，你是如何在一年之类取得社区成员的认可，成为这个社区的一员的呢？

多参与，混脸熟

**Comment by draveness** [comment]
> @chensheng0 大佬，想请问下：k8s pr 提了3天，没有人来 review，是正常的嘛？[kubernetes/kubernetes#95825](https://github.com/kubernetes/kubernetes/pull/95825)

挺正常的，你看下你 PR 的 Label 还有 CI 的一些问题

> @chensheng0: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our release note process to remove it.

**Comment by chensheng0** [comment]
想请问下：/triage accepted 这个 command 具体是什么意思，看了 document, 没理解具体场景和语意

**Comment by draveness** [comment]
> 想请问下：/triage accepted 这个 command 具体是什么意思，看了 document, 没理解具体场景和语意

可以看下 KEP https://github.com/kubernetes/enhancements/blob/master/keps/sig-contributor-experience/1553-issue-triage/README.md

**Comment by Cactusinhand** [comment]
@chensheng0  `triage` 本意是分诊的意思，就是进行鉴别分类，以便后续处理。
GitHub上项目的管理权限分为`Read` -> `Triage`->`Write`->`Maintain`->`Admin`, 所以triage处于普通人(只能读)和具有合入PR权限的人之间，作为管理社区特别是很活跃的社区的各种问题的一种方式。

**Comment by chensheng0** [comment]
大佬，想请教个问题，之前在使用 boltdb 的时候发现它是有把全局的大锁的，很多 goroutine 来进行 Tx 操作的时候争夺这把大锁，导致后来者延迟很严重（几秒级），想问下 etcd 底层使用的也是 bolddb，etcd 没有遇到这样的问题吗还是有什么优秀方案解决了？

**Comment by draveness** [comment]
> 大佬，想请教个问题，之前在使用 boltdb 的时候发现它是有把全局的大锁的，很多 goroutine 来进行 Tx 操作的时候争夺这把大锁，导致后来者延迟很严重（几秒级），想问下 etcd 底层使用的也是 bolddb，etcd 没有遇到这样的问题吗还是有什么优秀方案解决了？

我记得 etcd fork 开了个新分支，你可以看看修没修，我也不太清楚

**Comment by wxuedong** [comment]
老师你好 向Kubernetes SIGs 提交PR 跟Kubernetes 流程是一样的嚒 ？比如向Descheduler(https://github.com/kubernetes-sigs/descheduler) 这个提交PR 的流程是什么

**Comment by draveness** [comment]
> 老师你好 向Kubernetes SIGs 提交PR 跟Kubernetes 流程是一样的嚒 ？比如向Descheduler(https://github.com/kubernetes-sigs/descheduler) 这个提交PR 的流程是什么

跟正常的提交流程没啥区别，整个社区的规范都是一样的

**Comment by zpeng1997** [comment]
抱着大佬大腿成长

**Comment by weijiang1994** [comment]
大佬，怎么样才能像你一样那么叼 膜拜~

**Comment by sundongmin** [comment]
请问大佬, 如何debug k8s的源码啊? 比如执行kubectl exec时各种组件之间的调用

**Comment by draveness** [comment]
> 请问大佬, 如何debug k8s的源码啊? 比如执行kubectl exec时各种组件之间的调用

这个我也不太清楚，我开发的组件一般都只需要跑 go test，你可以看看相关的代码有没有测试用例

**Comment by jason-i-vv** [comment]
请问，怎么本地开发调试k8s的源码呢，比如是否可以本地启动API-server，连接服务器上的k8s环境,本地调试跟踪api的处理逻辑

**Reply by jason-i-vv** [reply]
已经解决了，源码可以本地运行，只需要保证x509签名证书包含127.0.0.1IP地址，就可以本地运行，链接服务器上的k8s环境了
